sql/_procedures.sql

```diff
@@ -22,43 +20,49 @@ CREATE OR REPLACE PROCEDURE correction (
     _uid text,
     _field text,
     _old_val text,
-    _new_val text
+    _new_val text,
+    _how_recode text
 ) AS $BODY$
 DECLARE
     field_type text;
-    pre_corr_val text;
-    applicable boolean;
+    records_to_recode boolean;
 BEGIN
     EXECUTE format($n$
         SELECT pg_typeof(a.%1$I) FROM %2$I a LIMIT 1;
     $n$, _field, _table) INTO field_type;
 
-    EXECUTE format($n$
-        SELECT a.%1$I::text FROM %2$I a WHERE a.uid = %3$L;
-    $n$, _field, _table, _uid) INTO pre_corr_val;
-
-    EXECUTE format($n$
-        SELECT %1$L::%3$s = %2$L::%3$s 
-        OR (%1$L IS NULL AND %2$L IS NULL)
-    $n$, pre_corr_val, _old_val, field_type) INTO applicable;
-
-    IF applicable THEN 
-        RAISE NOTICE 'Applying Correction';
+    IF _how_recode = 'id' THEN
+        EXECUTE format($n$
+            SELECT count(*) >0 FROM %1$I a WHERE a.uid = %2$L;
+        $n$, _table, _uid) INTO records_to_recode;
+    ELSE
         EXECUTE format($n$
-            UPDATE %1$I SET %2$I = %3$L::%4$s WHERE uid = %5$L;
-            $n$, _table, _field, _new_val, field_type, _uid);
+            SELECT count(*)>0 FROM %2$I a WHERE a.%1$I::text = %3$L;
+        $n$, _field, _table, _old_val) INTO records_to_recode;
+    END IF;
 
+    IF records_to_recode THEN 
+        RAISE NOTICE 'Applying Correction';
+        IF _how_recode = 'id' THEN
```

Comment:
Is this duplicative?  Should this be _how_recode = 'value' ?

Reply:
It's not duplicative, this second if/else block determines which recoding process to apply. The top if/else block based on `_how_recode` determines how to check whether there are valid values to recode